plotpal/volumes.py

```diff
@@ -0,0 +1,350 @@
+    """
+    
+        
+    if type(x_vals) == np.ndarray and type(y_vals) == np.ndarray :
+        yy, xx = np.meshgrid(y_vals, x_vals)
+        zz = z_vals * np.ones_like(xx)
+    elif type(x_vals) == np.ndarray and type(z_vals) == np.ndarray :
+        zz, xx = np.meshgrid(z_vals, x_vals)
+        yy = y_vals * np.ones_like(xx)
+    elif type(y_vals) == np.ndarray and type(z_vals) == np.ndarray :
+        zz, yy = np.meshgrid(z_vals, y_vals)
+        xx = x_vals * np.ones_like(yy)
+
+    if x_bounds is None:
+        if type(y_vals) == np.ndarray and type(z_vals) == np.ndarray and bool_function == np.logical_or :
+            x_bool = np.zeros_like(yy)
+        else:
+            x_bool = np.ones_like(yy)
+    else:
+        x_bool = (xx >= x_bounds[0])*(xx <= x_bounds[1])
+
+    if y_bounds is None:
+        if type(x_vals) == np.ndarray and type(z_vals) == np.ndarray and bool_function == np.logical_or :
+            y_bool = np.zeros_like(xx)
+        else:
+            y_bool = np.ones_like(xx)
+    else:
+        y_bool = (yy >= y_bounds[0])*(yy <= y_bounds[1])
+
+    if z_bounds is None:
+        if type(x_vals) == np.ndarray and type(y_vals) == np.ndarray and bool_function == np.logical_or :
+            z_bool = np.zeros_like(xx)
+        else:
+            z_bool = np.ones_like(xx)
+    else:
+        z_bool = (zz >= z_bounds[0])*(zz <= z_bounds[1])
+
+
+    side_bool = bool_function.reduce((x_bool, y_bool, z_bool))
+
+
+    side_info = OrderedDict()
+    side_info['x'] = np.where(side_bool, xx, np.nan)
+    side_info['y'] = np.where(side_bool, yy, np.nan)
+    side_info['z'] = np.where(side_bool, zz, np.nan)
+    side_info['surfacecolor'] = np.where(side_bool, data_vals, np.nan)
+
+    return side_info
+#turn colormesh into box with info on left side right side top, x, y and z basis, and colourmap
+class Box:
+    """
+    A struct containing information about a slice colormesh plot
+
+    # Attributes
+        task (str) :
+            The profile task name
+        x_basis, y_basis (strs) :
+            The dedalus basis names that the profile spans in the x- and y- direction of the plot
+        cmap  (str) :
+            The matplotlib colormap to plot the colormesh with
+        label (str):
+            A label for the colorbar
+
+    """
+
+    def __init__(self, left, right, top, x_basis='x', y_basis='y',z_basis='z', cmap='RdBu_r', \
+                              pos_def=False, \
+                              vmin=None, vmax=None, log=False, vector_ind=None, \
+                              label=None, cmap_exclusion=0.005):
+        
+        self.first=True
+        self.log=log
+        self.x_basis = x_basis
+        self.y_basis = y_basis
+        self.z_basis = z_basis
+        self.vector_ind = vector_ind
+        self.left=left
+        self.right=right
+        self.top=top
+
+
+        self.pos_def = pos_def
+        self.vmin = vmin
+        self.vmax = vmax
+        self.cmap_exclusion = cmap_exclusion
```

Comment:
Add a line here that if vmin != None and vmax != None, tell the user in a print statement that cmap_exclusion is being ignored.